tests/compilation_tests/grid_test.py

```diff
@@ -4,6 +4,7 @@
 from engine.grid import Grid
 from engine.kinematics import get_vincenty_x, get_vincenty_y
 from engine.mission import ControlMode
+from tests.functionality_tests.test_activation_shapes import is_on_border
```

Comment:
Since you removed the duplicate is_on_border, I think we can delete this import statement and use the remaining method from engine.grid (which is already imported)